### PR TITLE
Add galera_configuration and port_changed integration tests

### PR DESCRIPTION
**Pull request by sinfomicien**
- Include Port Changed PR (i canceled the first one with port_changed only tests)

### CODE REVIEW

File: libraries/helpers.rb [hunk 1]
```diff
@@ -168,8 +187,13 @@ def platform_service_name(_version = node.run_state['mariadb']['version'])
...
 
-    def mysql_command_string(database, query)
-      "psql -d #{database} <<< '#{query};'"
+    def restart_mariadb_service
+      # Using this to generate a service resource to control
+      service 'mariadb' do
+        service_name platform_service_name
+        supports restart: true, status: true, reload: true
+        action :restart
+      end
```
Comment (damacus) [inline libraries/helpers.rb:196]: We'll probably need this resource later on in the run, so why are we not making a generic copy of it somwher? 🤔

File: resources/server_configuration.rb [hunk 2]
```diff
@@ -104,65 +105,7 @@
...
-              mysqld_options: new_resource.mysqld_options,
-              mysqldump_quick: new_resource.mysqldump_quick,
-              mysqldump_quote_names: new_resource.mysqldump_quote_names,
-              mysqldump_max_allowed_packet: new_resource.mysqldump_max_allowed_packet,
-              mysqldump_options: new_resource.mysqldump_options,
-              isamchk_key_buffer: new_resource.isamchk_key_buffer,
-              isamchk_options: new_resource.isamchk_options,
-              extra_configuration_directory: new_resource.extra_configuration_directory
-             )
+    variables build_mycnf_variables_hash
```
Comment (damacus) [inline resources/server_configuration.rb]: 👎 this seems like unnecessary redirection and will decrease the readability of the cookbook.

**Review by damacus — CHANGES_REQUESTED** [review]

**Comment by damacus** [comment]
Hey @sinfomicien you've got conflicts in your kitchen files now :)

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: libraries/helpers.rb (hunk 1 above)
Comment (sinfomicien) [reply libraries/helpers.rb:196]: Because for now, we don't need it elsewhere 👿 ?

File: resources/server_configuration.rb (hunk 2 above)
Comment (sinfomicien) [reply resources/server_configuration.rb]: I remove that part of the code, following your recos

**Comment by sinfomicien** [comment]
Every comments has been taken into account :-D

**Review by damacus — DISMISSED** [review]

**Comment by damacus** [comment]
LGTM, you just have a few failing tests now :)

**Pull request merged**